src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -1373,19 +1375,21 @@ JNIEXPORT jint JNICALL
 Java_sun_nio_fs_UnixNativeDispatcher_fgetxattr0(JNIEnv* env, jclass clazz,
     jint fd, jlong nameAddress, jlong valueAddress, jint valueLen)
 {
-    size_t res = -1;
+    ssize_t res = -1;
     const char* name = jlong_to_ptr(nameAddress);
     void* value = jlong_to_ptr(valueAddress);
 
 #ifdef __linux__
     res = fgetxattr(fd, name, value, valueLen);
-#elif defined(_ALLBSD_SOURCE)
+#elif defined(MACOSX)
     res = fgetxattr(fd, name, value, valueLen, 0, 0);
+#elif defined(__FreeBSD__) || defined(__NetBSD__)
+    res = extattr_get_fd(fd, EXTATTR_NAMESPACE_USER, name, value, (size_t)valueLen);
 #else
     throwUnixException(env, ENOTSUP);
 #endif
 
-    if (res == (size_t)-1)
+    if (res == (ssize_t)-1)
```

Comment:
Is this casting really needed if you change type on res to ssize_t?

Reply:
No, it should not be.